### PR TITLE
[#156914616] feat: add payment metadata to message

### PR DESCRIPTION
**Pull request by gunzip**
... and forward payment amount + notice_number to the webhook (push notification)

we have to decide how to process this informations for emails

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#156914616](https://www.pivotaltracker.com/story/show/156914616): Il modello dati del messaggio va esteso con i metadati di pagamento

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: api/definitions.yaml [hunk 1]
```diff
@@ -97,8 +97,31 @@ NewMessage:
...
+    amount:
+      $ref: "#/PaymentAmount"
+    notice_number:
+      $ref: "#/PaymentNoticeNumber"
+PaymentNoticeNumber:
+  description: The field ["Numero Avviso"](https://pagopa-specifichepagamenti.readthedocs.io/it/latest/_docs/Capitolo7.html#il-numero-avviso-e-larchivio-dei-pagamenti-in-attesa)
+    of pagoPa, needed to identify the payment. Format is `<aux digit (1n)>[<application code> (2n)]<codice IUV (15|17n)>`.
+    See [pagoPa specs](https://www.agid.gov.it/sites/default/files/repository_files/specifiche_attuative_pagamenti_1_3_1_0.pdf) for more info on this field and the IUV.
+  type: string
+  minLength: 16
```
Comment (cloudify) [inline api/definitions.yaml]: would it make sense to have a regex for validation?

File: api/definitions.yaml [hunk 2]
```diff
@@ -97,8 +97,31 @@ NewMessage:
...
+  maxLength: 20
+PaymentAmount:
+  description: Amount of payment in euro cent.
+  type: integer
+  minimum: 1
+  maximum: 999999999999
```
Comment (cloudify) [inline api/definitions.yaml]: even if pagopa accepts these kind of value, I would stay in the safe side of things and limit the amount we could handle (e.g. to say, 1000 euros, or `100000` cents)

File: api/definitions.yaml [hunk 3]
```diff
@@ -162,6 +187,8 @@ CreatedMessageWithoutContent:
       $ref: "#/Timestamp"
     sender_service_id:
       type: string
+    payment_data:
```
Comment (cloudify) [inline api/definitions.yaml]: I would say that payment_data is a (structured) part of the message content, we are not going to store it unless the user enabled the inbox, so `MessageWithoutContent` will not include it.

File: api/definitions.yaml [hunk 4]
```diff
@@ -97,8 +97,31 @@ NewMessage:
       $ref: "#/MessageContent"
     default_addresses:
       $ref: "#/NewMessageDefaultAddresses"
+    payment_data:
```
Comment (cloudify) [inline api/definitions.yaml]: I would put `payment_data` inside `MessageContent`, they have the same lifecycle

File: api/definitions.yaml (hunk 3 above)
Comment (gunzip) [reply api/definitions.yaml]: moved under content

File: api/definitions.yaml (hunk 2 above)
Comment (gunzip) [reply api/definitions.yaml]: set to 100.000 cents = 1k eur

File: api/definitions.yaml (hunk 1 above)
Comment (gunzip) [reply api/definitions.yaml]: done (18 digits starting with either 0,1,2,3)

**Comment by cloudify** [comment]
looks good! thanks

**Pull request merged**